code-gen/src/main/scala/scalapb/grpc_web/GrpcServiceMetadataPrinter.scala

```diff
@@ -0,0 +1,277 @@
+      }
+      .outdent
+      .add("}")
+  }
+
+  private[this] val channel     = "_root_.io.grpc.Channel"
+  private[this] val callOptions = "_root_.io.grpc.CallOptions"
+
+  private[this] val abstractStub   = "_root_.io.grpc.stub.AbstractStub"
+  private[this] val streamObserver = "_root_.io.grpc.stub.StreamObserver"
+
+  private[this] val serverCalls = "_root_.io.grpc.stub.ServerCalls"
+  private[this] val clientCalls = "_root_.scalapb.grpc.ClientCalls"
+
+  private[this] val serverServiceDef = "_root_.io.grpc.ServerServiceDefinition"
+  private[this] val executionContext = "executionContext"
+  private[this] val metadata = "_root_.scalapb.grpc.grpcweb.Metadata.Metadata"
+
+  private[this] def clientMethodImpl(m: MethodDescriptor, blocking: Boolean) = {
+    def printCall(p: FunctionalPrinter) = {
+      val sig =
+        if (blocking) blockingMethodSignature(m, overrideSig = true) + " = {"
+        else serviceMethodSignature(m, overrideSig = true) + " = {"
+
+      val prefix = if (blocking) "blocking" else "async"
+
+      val methodName = prefix + (m.streamType match {
+        case StreamType.Unary           => "UnaryCall"
+        case StreamType.ServerStreaming => "ServerStreamingCall"
+        case StreamType.ClientStreaming => "ClientStreamingCall"
+        case StreamType.Bidirectional   => "BidiStreamingCall"
+      })
+
+      val args = Seq("channel", m.grpcDescriptor.nameSymbol, "options", "metadata") ++
+        (if (m.isClientStreaming) Seq() else Seq("request")) ++
+        (if ((m.isClientStreaming || m.isServerStreaming) && !blocking) Seq("responseObserver")
+        else Seq())
+
+      val body = s"${clientCalls}.${methodName}(${args.mkString(", ")})"
+      p.call(generateScalaDoc(m)).add(sig).addIndented(body).add("}").newline
+    }
+
+    PrinterEndo { p =>
+      m.streamType match {
+        case StreamType.Unary           => printCall(p)
+        case StreamType.ServerStreaming => printCall(p)
+        case _ => p
+      }
+    }
+
+  }
+
+  private def stubImplementation(
+                                  className: String,
+                                  baseClass: String,
+                                  methods: Seq[PrinterEndo]
+                                ): PrinterEndo = { p =>
+    val build =
+      s"override def build(channel: $channel, options: $callOptions): ${className} = new $className(channel, options)"
+    p.add(
+      s"class $className(channel: $channel, options: $callOptions = $callOptions.DEFAULT) extends $abstractStub[$className](channel, options) with $baseClass {"
+    )
+      .indent
+      .call(methods: _*)
+      .add(build)
+      .outdent
+      .add("}")
+  }
+
+
+  private[this] val stub: PrinterEndo = {
+    val methods = service.getMethods.asScala.map(clientMethodImpl(_, false)).toSeq
+    stubImplementation(service.stub, service.name, methods)
+  }
+
+  private[this] def methodDescriptor(method: MethodDescriptor) = PrinterEndo { p =>
```

Comment:
since we do our generation now, maybe all of this is no longer needed?

Reply:
I will address this yes it's not needed